.github/workflows/aks.yml

```diff
@@ -36,38 +45,46 @@ jobs:
           go-version: ${{ env.SETUP_GO_VERSION }}
       - name: Setup Ginkgo Test Framework
         run: go install github.com/onsi/ginkgo/ginkgo@v1.16.2
-      - name: Install Dependencies
-        run: |
-          choco install kubernetes-cli eksctl
 
-      - name: Configure AWS credentials
+      - name: Configure AWS credentials for Route53
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ github.event.inputs.aws_id || secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ github.event.inputs.aws_key || secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: us-east-2
-      - name: Create EKS cluster
+
+      # follow https://github.com/marketplace/actions/azure-login#configure-deployment-credentials
+      # az group create --name epinioCI --location westeurope
+      # az ad sp create-for-rbac --name "epinioCI" --sdk-auth --role contributor \
+      #   --scopes /subscriptions/{id}/resourceGroups/epinioC
+      - uses: azure/login@v1
+        with:
+          creds: ${{ github.events.inputs.azure_credentials || secrets.AZURE_CREDENTIALS }}
+
+      - name: Create AKS cluster
         id: create-cluster
+        shell: bash
         run: |
           id=$RANDOM
           echo '::set-output name=ID::'$id
-          eksctl create cluster --name=epinio-ci$id --region=us-east-2 --nodes=2 --node-type=t3.xlarge --node-volume-size=40 --managed --kubeconfig=kubeconfig-epinio-ci
+          az aks create --resource-group epinioCI --name epinioCI$id --node-count 1 --generate-ssh-keys
+          az aks get-credentials --resource-group epinioCI --name epinioCI$id --file kubeconfig
 
       - name: Installation Acceptance Tests
         env:
           REGISTRY_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
           REGISTRY_PASSWORD: ${{ secrets.DOCKERHUB_TOKEN }}
-          REGEX: Scenario4
+          REGEX: Scenario3
           AWS_ZONE_ID: ${{ secrets.AWS_ZONE_ID }}
-          AWS_DOMAIN: ${{ secrets.AWS_DOMAIN }}
+          AKS_DOMAIN: ${{ secrets.AKS_DOMAIN }}
+        shell: bash
         run: |
-          export KUBECONFIG=$PWD/kubeconfig-epinio-ci
+          export KUBECONFIG=$PWD/kubeconfig
           make test-acceptance-install
 
-      - name: Delete EKS cluster
-        if: ${{ always() }}
-        env:
-          KUBECONFIG: kubeconfig-epinio-ci
+      - name: Delete AKS cluster
+        if: ${{ always() && !github.event.inputs.keep_cluster }}
```

Comment:
What is the `always()` needed for ? IIUC correctly it is always true, and thus superfluous in the condition ?!

Reply:
I'm not sure if that's needed, but didn't invest another thirty minutes to find out. 

You need the `always()`, when you have no other condition, so that the job runs even for failed tests, i.e. for cleanup. I think it's save to drop it, as we still evaluate the `github.event.inputs.keep_cluster` expression. But expressions in Github Actions surprised me multiple times, so I feel we should test that change, because we don't want to keep clusters running.

`github.event.inputs.keep_cluster` is null for a scheduled run, and by default the empty string for a manually scheduled run. For a string null and empty string should both evaluate to false, according to the docs.

Reply:
I am ok with keeping it. Just confused. Maybe a quick note summarizing your explanation in the flow itself ?

Reply:
This one is needed: https://github.com/epinio/epinio/blob/main/.github/workflows/eks.yml#L66

Reply:
Agreed, based on your explanation above.